File: src/SkyBlock/enchants/sword/BaseMeleeEnchant.php
<?php


namespace SkyBlock\enchants\sword;

use pocketmine\event\entity\EntityDamageEvent;
use pocketmine\player\Player;
use SkyBlock\enchants\BaseEnchantment;

abstract class BaseMeleeEnchant extends BaseEnchantment {

    /**
     * @param Player $holder
     *
     * @return bool
     */
    public function isApplicableTo(Player $holder) : bool {
        return mt_rand(1, 25) === 1;
    }

    public function getLevel(int $level = 1) : int {
        return (int) (ceil($level / 2));
    }

    /**
     * @param Player            $player
     * @param Player            $attacker
     * @param EntityDamageEvent $ev
     * @param int               $enchantmentlevel
     */
    abstract public function onActivation(Player $player, Player $attacker, EntityDamageEvent $ev, int $enchantmentlevel) : void;
}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/BaseMeleeEnchant.java
package dev.skyblock.enchants;
import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
public abstract class BaseMeleeEnchant extends BaseEnchant {
    public BaseMeleeEnchant(int id, String name, int maxLevel) { super(id, name, maxLevel); }
    public abstract void onActivation(Player victim, Player attacker, EntityDamageByEntityEvent ev, int level);
    public abstract boolean isApplicableTo(Player holder, int level);
}
